lib/rucio/daemons/conveyor/fts_throttler.py

```diff
@@ -0,0 +1,429 @@
+                    t = FTS3Transfertool(rse_info[1])
+
+                    # extract FTS storage from dst-url
+                    tmp = rse['destination']['hits']['hits'][0]['_source']['payload']['dst-url'].split(':', 2)
+                    url = tmp[0] + ':' + tmp[1]
+
+                    n = rse['failure_ratio'].get('value')
+
+                    print('\033[91m RSE \033[0m' + rse_info[0] + '\033[91m on FTS host \033[0m' + rse_info[1] +
+                          '\033[91m has failure ratio \033[0m' + str(rse['failure_ratio'].get('value')) + '\033[91m on storage \033[0m' + url)
+                    logging.info(' RSE ' + rse_info[0] + ' on FTS host ' + rse_info[1] +
+                                 ' has failure ratio ' + str(rse['failure_ratio'].get('value')) + ' on storage ' + url)
+
+                    try:
+                        se = t.get_se_config(url)
+                        print(se)
+                    except KeyError:
+                        logging.warn('configuration for storage element was not found, config will be set from default values')
+                        # all FTS Host servers have a default reference storage named '*' that holds the default values for all storages that arent listed yet.
+                        default_storage = t.get_se_config('*')
+                        t.set_se_config(url, inbound_max_active=int((100 / (100 + n)) * default_storage['se_info']['inbound_max_active']),
+                                        outbound_max_active=int((100 / (100 + n)) * default_storage['se_info']['outbound_max_active']))
+
+                        logging.info(url + 'inbound_max_active changed from ' + str(default_storage['se_info']['inbound_max_active']) + ' to ' + str(int((100 / (100 + n)) * default_storage['se_info']['inbound_max_active'])) +
+                                     ', outbound_max_active changed from ' + str(default_storage['se_info']['outbound_max_active']) + ' to ' + str(int((100 / (100 + n)) * default_storage['se_info']['outbound_max_active'])))
+
+                        # cycle_info_dict is used to write changes down to the cycle file.
+                        cycle_info_dict['storages'].append({'storage': url, 'inbound_max_active': default_storage['se_info']['inbound_max_active'],
+                                                            'outbound_max_active': default_storage['se_info']['outbound_max_active'], 'failure_ratio': n,
+                                                            'tuned_inbound_max_active': int((100 / (100 + n)) * default_storage['se_info']['inbound_max_active']),
+                                                            'tuned_outbound_max_active': int((100 / (100 + n)) * default_storage['se_info']['outbound_max_active']),
+                                                            'fts-host': rse_info[1], 'time': str(datetime.datetime.now())})
+                        continue
+                    except Exception, error:
```

Comment:
Not py3k compatible : https://python-future.org/compatible_idioms.html#catching-exceptions